### DISCUSSION TITLE
What is the future plans for an OpenTofu Language Server ?

### DISCUSSION THREAD

**Discussion by gamunu** [opentofu/opentofu#1247] [General]
The Terraform Language Server (TLS) provides good support for development within popular code editors. However, achieving seamless integration with OpenTOFU's Program language currently requires configuration paths for all editors.

- Recognizing OpenTOFU: Natively identify and support OpenTOFU files (.tf), offering features like syntax highlighting, code completion, and diagnostics.
- Streamlined integration: Eliminate the need for workarounds, ensuring an effortless experience for developers working with OpenTOFU within their preferred editor.

**Comment by RLRabinowitz** [comment]
Hey @gamunu !

We have [an issue](https://github.com/opentofu/opentofu/issues/970) for supporting OpenTofu in the VS Code Language Server, and this issue has been accepted. It is frozen for now, mainly because currently there's no real differentiator in between OpenTofu and Terraform in terms of what's necessary from the Language Server. However, we are aware that down the line this will become more and more necessary for users of OpenTofu

Feel free to voice your opinion on that issue, and to track it our progress there in the future
